ocaml/xapi/xapi_clustering.ml

```diff
@@ -164,6 +164,27 @@ let with_clustering_lock_if_cluster_exists ~__context where f =
   | _ ->
       with_clustering_lock where f
 
+let find_cluster_host_by_network ~__context ~addr =
+  match
+    Db.Cluster_host.get_all ~__context
```

Comment:
This is called for each host in `nodel` below, while you have `all_cluster_hosts` already available. So you can avoid all those DB calls by passing `all_cluster_hosts` to this function as an argument.